content/topics/Analyze/causal-inference/introduction/synthetic-control-methods-intro.md

```diff
@@ -0,0 +1,207 @@
+
+## Implementation in R 
+
+The implementation steps follow [this](https://github.com/edunford/tidysynth/blob/master/README.md). 
+
+The package we are going to be using is called *tidysynth*, with functionalities specific for the implementation of this method. 
+
+Let's start with installing and loading the package. 
+
+{{% codeblock %}}
+```R
+install.packages('tidysynth')
+library(tidysynth)
+```
+{{% /codeblock %}}
+
+The dataset we are going to use in this tutorial is the *smoking* dataset, based on the paper by Abadie et al. (2010), which evaluates the impact of tobacco control program on cigarette consumption in California.
+
+{{% codeblock %}}
+```R
+require(tidysynth)
+data("smoking")
+View(smoking)
+```
+{{% /codeblock %}}
+
+The method aims to generate a synthetic California using information from a subset of control states (the “donor pool”) where a similar law was not implemented. The donor pool is the subset of case comparisons from which information is borrowed to generate a synthetic version of the treated unit (“California”). 
+
+{{% codeblock %}}
+```R
+smoking_out <-
+  
+  smoking %>%
+  
+  # initiate the synthetic control object
+  synthetic_control(outcome = cigsale, # outcome
+                    unit = state, # unit index in the panel data
+                    time = year, # time index in the panel data
+                    i_unit = "California", # unit where the intervention occurred
+                    i_time = 1988, # time period when the intervention occurred
+                    generate_placebos=T # generate placebo synthetic controls (for inference)
+                    ) %>%
+  # Generate the aggregate predictors used to fit the weights (in our example referred to as W')
+  
+  # average log income, retail price of cigarettes, and proportion of the
+  # population between 15 and 24 years of age from 1980 - 1988
+  generate_predictor(time_window = 1980:1988,
+                     ln_income = mean(lnincome, na.rm = T),
+                     ret_price = mean(retprice, na.rm = T),
+                     youth = mean(age15to24, na.rm = T)) %>%
+  # average beer consumption in the donor pool from 1984 - 1988
+  generate_predictor(time_window = 1984:1988,
+                     beer_sales = mean(beer, na.rm = T)) %>%
+  
+  # Lagged cigarette sales 
+  generate_predictor(time_window = 1975,
+                     cigsale_1975 = cigsale) %>%
+  generate_predictor(time_window = 1980,
+                     cigsale_1980 = cigsale) %>%
+  generate_predictor(time_window = 1988,
+                     cigsale_1988 = cigsale) %>%
+  
+   # Generate the fitted weights for the synthetic control
+  generate_weights(optimization_window = 1970:1988, # time to use in the optimization task
+                   margin_ipop = .02,sigf_ipop = 7,bound_ipop = 6 # optimizer options
+  ) %>%
+  
+  # Generate the synthetic control
+  generate_control()
+```
+{{% /codeblock %}}
```

Comment:
Is there a possibility to use more code blocks for this code? And maybe add the intuition in text-form instead of comments with #? I can't read the full comment without scrolling the big code block (maybe it's my laptop). I would keep the comments in the code with # as short as possible or otherwise divide them over two lines, so you don't have to scroll too much! 

And, is "aggregate predictor" the same as the weighting vector W? Maybe use the same word for that!